diff --git a/firefox-ios/firefox-ios-tests/Tests/ClientTests/OnboardingTests/NimbusOnboardingFeatureLayerTests.swift b/firefox-ios/firefox-ios-tests/Tests/ClientTests/OnboardingTests/NimbusOnboardingFeatureLayerTests.swift
index 2b888ca051..9d7aed0893 100644
--- a/firefox-ios/firefox-ios-tests/Tests/ClientTests/OnboardingTests/NimbusOnboardingFeatureLayerTests.swift
+++ b/firefox-ios/firefox-ios-tests/Tests/ClientTests/OnboardingTests/NimbusOnboardingFeatureLayerTests.swift
@@ -2,9 +2,11 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/
 
+import Common
 import MozillaAppServices
-import XCTest
 import OnboardingKit
+import Shared
+import XCTest
 
 @testable import Client
 
@@ -19,14 +21,16 @@ class NimbusOnboardingFeatureLayerTests: XCTestCase {
     }
 
     override func tearDown() {
-        configUtility = nil
         super.tearDown()
+        configUtility = nil
     }
 
     // MARK: - Test placeholder methods
     func testLayer_placeholderNamingMethod_returnsExpectedStrigs() {
         setupNimbusForStringTesting()
-        let expectedPlaceholderString = "A string inside Firefox with a placeholder"
+        // Ecosia: Update placeholder
+        // let expectedPlaceholderString = "A string inside Firefox with a placeholder"
+        let expectedPlaceholderString = "A string inside Ecosia with a placeholder"
         let expectedNoPlaceholderString = "On Wednesday's, we wear pink"
         let layer = NimbusOnboardingFeatureLayer(with: MockNimbusMessagingHelperUtility())
 
@@ -100,19 +104,18 @@ class NimbusOnboardingFeatureLayerTests: XCTestCase {
             body: CardElementNames.body + " 1",
             link: OnboardingLinkInfoModel(title: CardElementNames.linkTitle,
                                           url: URL(string: CardElementNames.linkURL)!),
-            buttons: OnboardingButtons<OnboardingActions>(
-                primary: OnboardingButtonInfoModel<OnboardingActions>(
+            buttons: OnboardingButtons(
+                primary: OnboardingButtonInfoModel(
                     title: CardElementNames.primaryButtonTitle,
                     action: .forwardOneCard),
-                secondary: OnboardingButtonInfoModel<OnboardingActions>(
+                secondary: OnboardingButtonInfoModel(
                     title: CardElementNames.secondaryButtonTitle,
                     action: .forwardOneCard)),
             multipleChoiceButtons: [],
             onboardingType: .freshInstall,
             a11yIdRoot: CardElementNames.a11yIDOnboarding,
             imageID: ImageIdentifiers.Onboarding.HeaderImages.welcomev106,
-            instructionsPopup: nil,
-            embededLinkText: [])
+            instructionsPopup: nil)
 
         XCTAssertEqual(subject.name, expectedCard.name)
         XCTAssertEqual(subject.title, expectedCard.title)
