diff --git a/firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarActionConfiguration.swift b/firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarActionConfiguration.swift
index 7aefc9d8ea..1492685d4a 100644
--- a/firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarActionConfiguration.swift
+++ b/firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarActionConfiguration.swift
@@ -10,6 +10,8 @@ struct ToolbarActionConfiguration: Equatable, FeatureFlaggable {
         case back
         case forward
         case home
+        // Ecosia: History action triggered from the NTP toolbar button
+        case history
         case newTab
         case search
         case tabs
@@ -30,6 +32,11 @@ struct ToolbarActionConfiguration: Equatable, FeatureFlaggable {
     var actionLabel: String?
     var iconName: String?
     var badgeImageName: String?
+    // Ecosia: Badge customisation parameters for the incognito icon
+    var badgeBundle: Bundle?
+    var badgeSize: CGSize?
+    var badgeXOffset: CGFloat?
+    var badgeYOffset: CGFloat?
     var maskImageName: String?
     var templateModeForImage = true
     var loadingConfig: LoadingConfig?
