diff --git a/firefox-ios/firefox-ios-tests/Tests/ClientTests/OverlayModeManagerTests.swift b/firefox-ios/firefox-ios-tests/Tests/ClientTests/OverlayModeManagerTests.swift
index 3795a5fd59..53be355092 100644
--- a/firefox-ios/firefox-ios-tests/Tests/ClientTests/OverlayModeManagerTests.swift
+++ b/firefox-ios/firefox-ios-tests/Tests/ClientTests/OverlayModeManagerTests.swift
@@ -5,6 +5,8 @@
 import XCTest
 
 @testable import Client
+
+@MainActor
 class OverlayModeManagerTests: XCTestCase {
     private var urlBar: MockURLBarView!
     private var subject: MockOverlayModeManager!
@@ -16,13 +18,12 @@ class OverlayModeManagerTests: XCTestCase {
     }
 
     override func tearDown() {
+        super.tearDown()
         urlBar = nil
         subject = nil
-        super.tearDown()
     }
 
     // MARK: - Test URLBarView nil
-    @MainActor
     func testOverlayMode_ForNilURLBar() {
         urlBar = nil
         subject.openSearch(with: "search")
@@ -30,26 +31,30 @@ class OverlayModeManagerTests: XCTestCase {
     }
 
     // MARK: - Test EnterOverlay for New tab
-    @MainActor
-    func testEnterOverlayMode_ForNewTabHome_WithNilURL() {
+    // Ecosia: Update test name according to the expected behaviour
+    // func testEnterOverlayMode_ForNewTabHome_WithNilURL() {
+    func testDoesNotEnterOverlayMode_ForNewTabHome_WithNilURL() {
         subject.setURLBar(urlBarView: urlBar)
         subject.openNewTab(url: nil, newTabSettings: .topSites)
-
-        XCTAssertTrue(subject.inOverlayMode)
+        // Ecosia: Update with the new Ecosia's behaviour where overlay mode is OFF for New Tab
+        // XCTAssertTrue(subject.inOverlayMode)
+        XCTAssertFalse(subject.inOverlayMode)
         XCTAssertEqual(subject.enterOverlayModeCallCount, 1)
     }
 
-    @MainActor
-    func testEnterOverlayMode_ForNewTabHome_WithHomeURL() {
+    // Ecosia: Update test name according to the expected behaviour
+    // func testEnterOverlayMode_ForNewTabHome_WithHomeURL() {
+    func testDoesNotEnterOverlayMode_ForNewTabHome_WithHomeURL() {
         subject.setURLBar(urlBarView: urlBar)
         subject.openNewTab(url: URL(string: "internal://local/about/home"),
                            newTabSettings: .topSites)
 
-        XCTAssertTrue(subject.inOverlayMode)
+        // Ecosia: Update with the new Ecosia's behaviour where overlay mode is OFF for New Tab
+        // XCTAssertTrue(subject.inOverlayMode)
+        XCTAssertFalse(subject.inOverlayMode)
         XCTAssertEqual(subject.enterOverlayModeCallCount, 1)
     }
 
-    @MainActor
     func testEnterOverlayMode_ForNewTabHome_WithURL() {
         subject.setURLBar(urlBarView: urlBar)
         subject.openNewTab(url: URL(string: "https://test.com"),
@@ -59,7 +64,6 @@ class OverlayModeManagerTests: XCTestCase {
         XCTAssertEqual(subject.enterOverlayModeCallCount, 1)
     }
 
-    @MainActor
     func testEnterOverlayMode_ForBlankPage_WithNilURL() {
         subject.setURLBar(urlBarView: urlBar)
         subject.openNewTab(url: nil, newTabSettings: .blankPage)
@@ -68,7 +72,6 @@ class OverlayModeManagerTests: XCTestCase {
         XCTAssertEqual(subject.enterOverlayModeCallCount, 1)
     }
 
-    @MainActor
     func testEnterOverlayMode_ForBlankPage_WithURL() {
         subject.setURLBar(urlBarView: urlBar)
         subject.openNewTab(url: URL(string: "https://test.com"),
@@ -78,7 +81,6 @@ class OverlayModeManagerTests: XCTestCase {
         XCTAssertEqual(subject.enterOverlayModeCallCount, 1)
     }
 
-    @MainActor
     func testNotEnterOverlayMode_ForCustomUrl() {
         subject.setURLBar(urlBarView: urlBar)
         subject.openNewTab(url: URL(string: "https://test.com"),
@@ -87,7 +89,6 @@ class OverlayModeManagerTests: XCTestCase {
         XCTAssertFalse(subject.inOverlayMode)
     }
 
-    @MainActor
     func testNotEnterOverlayMode_ForCustomUrl_WithNilURL() {
         subject.setURLBar(urlBarView: urlBar)
         subject.openNewTab(url: nil,
@@ -98,7 +99,6 @@ class OverlayModeManagerTests: XCTestCase {
 
     // MARK: - Test EnterOverlay for paste action
 
-    @MainActor
     func testEnterOverlayMode_ForPasteContent() {
         subject.setURLBar(urlBarView: urlBar)
         subject.openSearch(with: "paste")
@@ -109,7 +109,6 @@ class OverlayModeManagerTests: XCTestCase {
 
     // MARK: - Test EnterOverlay for finish editing
 
-    @MainActor
     func testLeaveOverlayMode_ForFinishEditing() {
         subject.setURLBar(urlBarView: urlBar)
         subject.finishEditing(shouldCancelLoading: true)
@@ -119,7 +118,6 @@ class OverlayModeManagerTests: XCTestCase {
     }
 
     // MARK: - Test EnterOverlay for Tab change
-    @MainActor
     func testLeaveOverlayMode_ForSwitchTab() {
         subject.setURLBar(urlBarView: urlBar)
         subject.switchTab(shouldCancelLoading: true)
@@ -128,7 +126,6 @@ class OverlayModeManagerTests: XCTestCase {
         XCTAssertEqual(subject.leaveOverlayModeCallCount, 1)
     }
 
-    @MainActor
     func testLeaveOverlayMode_ForSwitchTabInOverlayMode() {
         subject.setURLBar(urlBarView: urlBar)
         subject.openSearch(with: "")
